cmd/entrypoint/post_writer.go

```diff
@@ -13,11 +14,16 @@ type realPostWriter struct{}
 var _ entrypoint.PostWriter = (*realPostWriter)(nil)
 
 // Write creates a file and writes content to that file if content is specified
-// assumption here is the underlying directory structure already exists
 func (*realPostWriter) Write(file string, content string) {
 	if file == "" {
 		return
 	}
+
+	// Create directory if it doesn't already exist
+	if err := os.MkdirAll(filepath.Dir(file), os.ModePerm); err != nil {
+		log.Fatalf("Error creating parent directory of %q: %v", file, err)
+	}
+
```

Comment:
Could you add tests for this new behaviour?
I tried removing this and unit tests in `cmd/entrypoint/post_writer_test.go` still work fine.